xauc/metrics.py

```diff
@@ -0,0 +1,107 @@
+# Lint as: python3
+"""Implementations of the xAUC metrics.
+
+Implementations of the Cross-Area Under the Curve metric and its balanced
+variants, xAUC_0 and xAUC_1.
+
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+import numpy as np
+from sklearn import metrics
+
+import probability
+
+
+def x_auc(scores_a_desired, scores_b_undesired, desired_label, undesired_label):
+    """Returns the Cross-Area Under Curve (xAUC) metric of the given scores.
+
+    The xAUC metric is defined as the probability "desired" examples in class a
```

Comment:
I am just thinking if we should use other terms like 'protected group'  etc instead of desired. We can discuss in more detail offline.

Reply:
Totally!